.github/workflows/ci_release.yml

```diff
@@ -0,0 +1,69 @@
+name: CI and Release
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+    # Inputs the workflow accepts.
+    inputs:
+      version:
+        # Friendly description to be shown in the UI instead of 'name'
+        description: "Semver type of new version (major / minor / patch)"
+        # Input has to be provided for the workflow to run
+        required: true
+        type: choice
+        options:
+          - patch
+          - minor
+          - major
+
+jobs:
+  # Dockerfile Linting
+  hadolint:
+    uses: celestiaorg/.github/.github/workflows/reusable_dockerfile_lint.yml@main # yamllint disable-line rule:line-length
+    with:
+      dockerfile: docker/Dockerfile
+
+  yamllint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: celestiaorg/.github/.github/actions/yamllint@main
+
+  markdown-lint:
+    name: Markdown Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: |
+          npm install -g markdownlint-cli@0.32.1
+          markdownlint --config .markdownlint.yaml '**/*.md'
+
+  go-ci:
+    uses: ./.github/workflows/go-ci.yml
+
+  # Make a release if this is a manually trigger job, i.e. workflow_dispatch
+  release:
+    needs: [hadolint, yamllint, markdown-lint, go-ci]
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'workflow_dispatch' }}
+    permissions: "write-all"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Version Release
+        uses: celestiaorg/.github/.github/actions/version-release@main
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          version-bump: ${{inputs.version}}
+
+  # TODO: permission issue, but not worth fixing as this should be refactored
+  # into the celestiaorg/.github repo, at which point any permission issues will
+  # be resolved.
```

Comment:
I saw a lot of progress on `.github` repo. When do we migrate?

Reply:
As soon as we want. It would be good to make that a follow up after this PR is merged to begin refactoring more. 

Some can be refactored immediate, others it makes sense to coordinate with the other teams first.